cmd/load.go

```diff
@@ -0,0 +1,286 @@
+		go func() {
+			// cancelChannel will be notified when os receives os.Interrupt and os.Kill signals
+			signal.Notify(cancelChannel, os.Interrupt, os.Kill)
+
+			for {
+				select {
+				case <-cancelChannel:
+					glcm.Info("Cancellation requested. Beginning shutdown...")
+					return
+				}
+			}
+		}()
+
+		err = clfscmd.Start()
+		if err != nil {
+			glcm.Error("Cannot start job due to error: " + err.Error())
+		}
+
+		clfsOutputParser.finishParsing()
+
+		err = clfscmd.Wait()
+		if err != nil {
+			glcm.Error("Job failed due to error: " + err.Error())
+		}
+
+		glcm.Exit(func(format common.OutputFormat) string {
+			return ""
+		}, common.EExitCode.Success())
+	},
+}
+
+func init() {
+	rootCmd.AddCommand(loadCmd)
+	loadCmd.PersistentFlags().BoolVar(&loadCmdRawInput.newSession, "new-session", true, "start a new job rather than continuing an existing one whose tracking information is kept at --state-path.")
+	loadCmd.PersistentFlags().StringVar(&loadCmdRawInput.statePath, "state-path", "", "required path to a local directory for job state tracking. The path must point to an existing directory in order to resume a job.")
+	loadCmd.PersistentFlags().StringVar(&loadCmdRawInput.compression, "compression-type", "LZ4", "specify the compression type to use for the transfers.")
+	loadCmd.PersistentFlags().Uint32Var(&loadCmdRawInput.maxErrorsToTolerate, "max-errors", 0, "specify the maximum number of transfer failures to tolerate. If enough errors occur, stop the job immediately.")
+	loadCmd.PersistentFlags().BoolVar(&loadCmdRawInput.preserveHardlinks, "preserve-hardlinks", false, "preserve hard link relationships.")
+	loadCmd.PersistentFlags().StringVar(&loadCmdRawInput.logLevel, "log-level", "INFO", "define the log verbosity for the log file, available levels: DEBUG, INFO, WARNING, ERROR.")
+	loadCmd.PersistentFlags().Uint32Var(&loadCmdRawInput.numConcurrentWorkers, "concurrency-count", 0, "override the number of parallel connections.")
+	loadCmd.PersistentFlags().MarkHidden("concurrency-count")
+}
+
+func getClfsExtensionPathAndVerifyHash(expectedHash string) (string, error) {
+	currentParentDir, err := filepath.Abs(filepath.Dir(os.Args[0]))
+	if err != nil {
+		glcm.Error(err.Error())
+	}
+
+	clfsToolPath := filepath.Join(currentParentDir, clfsToolName)
+	_, err = os.Stat(clfsToolPath)
+
+	// in case the CLFS extension is not next to our azcopy executable, find it under the user's PATH
+	if err != nil {
+		clfsToolPath, err = exec.LookPath(clfsToolName)
+		if err != nil {
+			return "", errors.New("cannot find CLFSLoad extension, please put it next to AzCopy, or anywhere under your PATH")
+		}
+	}
+
+	file, err := os.Open(clfsToolPath)
+	if err != nil {
+		return "", errors.New("cannot open CLFSLoad extension")
+	}
+	defer file.Close()
+
+	//make sure we will be invoking the right CLFSLoad
+	hash := md5.New()
+	if _, err := io.Copy(hash, file); err != nil {
+		return clfsToolPath, errors.New("cannot compute hash of CLFSLoad extension")
+	}
+
```

Comment:
Consider adding a hidden debug flag to substitute the clfsload executable with an alternate one, in case there's say, an update and they want to test it.